test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -25,994 +43,1415 @@ const noop = () => {};
-          // #. Using ``client``, drop the collections ``keyvault.datakeys`` and ``db.coll``.
-          .then(() => dropCollection(this.client.db(dataDbName), dataCollName))
-          .then(() => dropCollection(this.client.db(keyVaultDbName), keyVaultCollName))
-          // #. Create the following:
-          //   - A MongoClient configured with auto encryption (referred to as ``client_encrypted``)
-          //   - A ``ClientEncryption`` object (referred to as ``client_encryption``)
-          //   Configure both objects with ``aws`` and the ``local`` KMS providers as follows:
-          //   .. code:: javascript
-          //       {
-          //           "aws": { <AWS credentials> },
-          //           "local": { "key": <base64 decoding of LOCAL_MASTERKEY> }
-          //       }
-          //   Configure both objects with ``keyVaultNamespace`` set to ``keyvault.datakeys``.
-          //   Configure the ``MongoClient`` with the following ``schema_map``:
-          //   .. code:: javascript
-          //       {
-          //         "db.coll": {
-          //           "bsonType": "object",
-          //           "properties": {
-          //             "encrypted_placeholder": {
-          //               "encrypt": {
-          //                 "keyId": "/placeholder",
-          //                 "bsonType": "string",
-          //                 "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
-          //               }
-          //             }
-          //           }
-          //         }
-          //       }
-          //   Configure ``client_encryption`` with the ``keyVaultClient`` of the previously created ``client``.
-          .then(() => {
-            this.clientEncryption = new mongodbClientEncryption.ClientEncryption(this.client, {
-              kmsProviders: getKmsProviders(),
-              keyVaultNamespace
+      before(async function () {
+        client = this.configuration.newClient();
+        await client.connect();
+        const mongodbClientEncryption = this.configuration.mongodbClientEncryption;
+        clientEncryption = new mongodbClientEncryption.ClientEncryption(client, {
+          ...autoEncryptionOptions,
+          bson: BSON
+        });
+        clientEncryptionInvalid = new mongodbClientEncryption.ClientEncryption(client, {
+          ...autoEncryptionOptions,
+          bson: BSON,
+          kmsProviders: {
+            kmip: {
+              endpoint: 'invalid.localhost:5698'
+            }
+          }
+        });
+        await dropCollection(client.db(keyVaultDbName), keyVaultCollName);
+        await dropCollection(client.db(keyVaultDbName), keyVaultCollName);
+      });
+
+      after(async function () {
+        await client.close();
+      });
+
+      context('when encrypting with kmip', metadata, async function () {
+        context('when not providing an endpoint in the master key', metadata, async function () {
+          const masterKey = { keyId: '1' };
```

Comment:
The spec says that keyId is always 96 bytes, shouldn't this be an issue?

Reply:
I'm not sure why the prose tests had the keyId as a string but it does work for some reason. I went back to the spec again and I see no mention of what would happen if this isn't a 96 byte UUID. Also went back to the original PR for it on the spec repo and there are no comments around it.

Reply:
I'm trying to look more into this, cus the behavior also states that if the key isn't provided the driver is supposed to fill in random 96 bytes. I want to make sure we're not filling in random bytes over this '1' if it somehow allowed. Can you link me to the the prose test? I can't seem to find it.